train.py

```diff
@@ -219,17 +219,16 @@ def loss_fn(pred, labels):
 
     metric_logger = build_metric_logger(job_config)
 
-    # torch.compile model for improved performance
     if job_config.training.compile:
         if (
             job_config.activation_checkpoint.mode == "selective"
             and job_config.activation_checkpoint.selective_ac_option == "op"
         ):
+            # some flags for torch.compile enablement
             torch._dynamo.config._experimental_support_context_fn_in_torch_utils_checkpoint = (
                 True
             )
-        logger.info("Compiling model with torch.compile")
-        model = torch.compile(model)
+        logger.info("Compiling each TransformerBlock with torch.compile")
```

Comment:
nit: Since we put parallelization, ac, and compile all in parallelize_llama.py, it might make more sense to put this all in that file. This will simplify `train.py`.

Concretely, we can put this block to `parallelize_llama.py`, right before `logger.info("Applied FSDP to the model")` and change the wording from `"Compiling each ..."` to `"Compiled each ..."`